src/server/dfly_main.cc

```diff
@@ -479,7 +495,13 @@ Usage: dragonfly [FLAGS]
   sigemptyset(&act.sa_mask);
   sigaction(SIGILL, &act, nullptr);
 
-  CHECK_GT(GetFlag(FLAGS_port), 0u);
+  if (GetFlag(FLAGS_port) == 0u) {
+    auto usock = GetFlag(FLAGS_unixsocket);
+    CHECK_GT(usock.length(), 0u);
```

Comment:
Instead of CHECK_GT, please write something "either --port or --unixsocket must be enabled, exiting..".

Reply:
This is handled later, inside `RunEngine`, should this logic move to main?

https://github.com/dragonflydb/dragonfly/blob/4657f993534a4c51d4c1945e08a23bee25c8f01f/src/server/dfly_main.cc#L347-L349

Reply:
If `--port=0` is specified, will dragonfly check-fail here? if yes, then obviously handling later does not help here.